app/models/generic_object_definition.rb

```diff
@@ -6,7 +6,11 @@ class GenericObjectDefinition < ApplicationRecord
   has_one   :picture, :dependent => :destroy, :as => :resource
   has_many  :generic_objects
 
-  def defined_attributes
-    @defined_attributes ||= properties[:attributes].stringify_keys
+  def defined_property_attributes
+    @defined_property_attributes ||= properties[:attributes].stringify_keys
```

Comment:
I was concerned about this in the previous PR, but is the memoization even needed here?  If you store the attributes in the column with a specific and consistent key type (either symbol or string), then it is not expensive to look it up in `property_attribute_defined?`